collectors/jiraffe/core.py

```diff
@@ -39,125 +37,120 @@ def _get_jira_connection(server, token) -> JIRA:
     def jira_conn(self) -> JIRA:
         return self._get_jira_connection(self._jira_server, self._jira_token)
 
-    ########################
-    # SINGLE ISSUE QUERIES #
-    ########################
-
-    def get_issue(self, jira_id):
-        """get Jira issue"""
-        return self.jira_conn.issue(jira_id)
-
 
-@cache
-def get_jira_modules() -> dict:
-    """
-    Returns a dict with all ps_modules/ps_components tracked in JIRA.
-    """
-    # TODO: need a proper/generic solution that provides product definitions in a reliable
-    # and efficient way, as it stands this will never be updated in long-running OSIDB instances
-    response = requests.get(
-        PRODUCT_DEFINITIONS_URL,
-        timeout=settings.DEFAULT_REQUEST_TIMEOUT,
-    )
-    if not response.ok:
-        return {}
-    data = response.json()
-    modules = data["ps_modules"]
-    return {
-        key: value for key, value in modules.items() if value["bts"]["name"] == "jboss"
-    }
-
-
-def find_jira_trackers(affect: Affect) -> list[Issue]:
+class JiraQuerier(JiraConnector):
     """
-    Finds JIRA trackers pertaining to a given affect.
+    Jira query handler
     """
-    jira_modules = get_jira_modules()
-    module = jira_modules.get(affect.ps_module)
-    if module is None:
-        return []
 
-    conn = JiraQuerier().jira_conn
-    jira_project = module["bts"]["key"]
-    jql_query = f'PROJECT={jira_project} \
-                AND labels="{affect.flaw and affect.flaw.cve_id}" \
-                AND labels="pscomponent:{affect.ps_component}" \
-                AND labels="SecurityTracking" \
-                AND type="Bug"'
-    return conn.search_issues(jql_query, maxResults=False)
+    ###########
+    # HELPERS #
+    ###########
 
+    def datetime2jira_str(self, timestamp: timezone.datetime, inc: bool = False) -> str:
```

Comment:
I would expect the JIRA lib to convert datetimes to the right format, is it not the case?

Reply:
The thing is that `search_issues` expects a string (https://jira.readthedocs.io/api.html#jira.client.JIRA.search_issues). Another reason of this method is the rounding which is more related to the defining the sync window.